uvtools/plot.py

```diff
@@ -2,115 +2,385 @@
 import numpy as np
 
 def data_mode(data, mode='abs'):
-    if mode.startswith('phs'): data = np.angle(data)
+    """
+    Apply filter to data according to a chosen plotting mode.
+    
+    Parameters
+    ----------
+    data : array_like
+        Array of data to be plotted (normally complex floats).
+    
+    mode : str, optional
+        Which transformation to apply to the data. Options are: 
+          - 'phs':  Phase angle.
+          - 'abs':  Absolute value.
+          - 'real': Real value.
+          - 'imag': Imaginary value.
+          - 'log':  Log (base-10) of absolute value.
+        Default: 'abs'.
+    
+    Returns
+    -------
+    data : array_like
+        Data transformed according to the value of `mode`.
+    """
+    if mode.startswith('phs'):
+        data = np.angle(data)
     elif mode.startswith('abs'):
         data = np.absolute(data)
         #data = np.masked_less_equal(data, 0)
-    elif mode.startswith('real'): data = data.real
-    elif mode.startswith('imag'): data = data.imag
+    elif mode.startswith('real'):
+        data = data.real
+    elif mode.startswith('imag'):
+        data = data.imag
     elif mode.startswith('log'):
         data = np.absolute(data)
         data = np.log10(data)
-    else: raise ValueError('Unrecognized plot mode.')
+    else:
+        raise ValueError('Unrecognized plot mode.')
     return data
 
-def waterfall(d, mode='log', mx=None, drng=None, recenter=False, **kwargs):
+
+def waterfall(d, mode='log', vmin=None, vmax=None, drng=None, mx=None, 
+              recenter=False, **kwargs):
+    """
+    Generate a 2D waterfall plot.
+    
+    Parameters
+    ----------
+    d : array_like
+        2D array of data.
+    
+    mode : str, optional
+        Which transformation to apply to the data before plotting. Options are: 
+          - 'phs':  Phase angle.
+          - 'abs':  Absolute value.
+          - 'real': Real value.
+          - 'imag': Imaginary value.
+          - 'log':  Log (base-10) of absolute value.
+        Default: 'log'.
+    
+    vmin, vmax : float, optional
+        Minimum and maximum values of the color scale. If not set (and `mx` and 
+        `drng` are not set), the min. and max. values of the data will be used.
+        
+        Note that that the min. and max. values are the ones _after_ the data 
+        have been transformed according to `mode`. So, if `mode='log'`, these 
+        values are the min. and max. of log_10(data).
+        
+    mx : float, optional (deprecated)
```

Comment:
It seems odd that this and `drng` are said to be deprecated, but they are still working options. In fact, the code internally uses `mx` and `drng` even if `vmin` and `vmax` are provided.

Reply:
I've removed the "deprecated" language. This function is widely used in many packages, so I wanted to maintain backwards compatibility.